### PR TITLE
Deprecating Academy.Instance.FloatProperties

### PR DESCRIPTION
**Pull request by vincentpierre**
### Proposed change(s)

 - Introduce the `SideChannelUtils` static class that maintains a static dictionary from GUID to SideChannel. This `SideChannelUtils` is the entry point for registering/unregistering/get Side Channlels
 - Replace `Academy.Instance.FloatProperties` with `SideChannelUtils.GetSideChannel<FloatPropertiesChannel>()` 

### Useful links (Github issues, JIRA tickets, ML-Agents forum threads etc.)
[JIRA MLA-799](https://jira.unity3d.com/browse/MLA-799)

### Types of change(s)

- ~[ ] Bug fix~
- ~[ ] New feature~
- [x] Code refactor
- [x] Breaking change
- ~[ ] Documentation update~
- ~[ ] Other (please describe)~

### Checklist
- ~[ ] Added tests that prove my fix is effective or that my feature works~
- [x] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [x] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [x] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### CODE REVIEW

File: com.unity.ml-agents/Runtime/Academy.cs [hunk 1]
```diff
@@ -89,6 +89,7 @@ public static bool IsInitialized
         /// <summary>
         /// Collection of float properties (indexed by a string).
         /// </summary>
+        [Obsolete("Academy.FloatProperties has been deprecated, use Academy.GetSideChannel<>() instead.")]
```
Comment (chriselion) [inline com.unity.ml-agents/Runtime/Academy.cs]: Should we just remove it? Otherwise we'll be stuck with it for all of 1.x

**Comment by chriselion** [comment]
General comment: since it's possible to have multiple FloatPropertiesChannels, should we at least one example of using `Academy.Instance.GetSideChannels<FloatPropertiesChannel>()[0]`?

**Comment by vincentpierre** [comment]
>General comment: since it's possible to have multiple FloatPropertiesChannels, should we at least one example of using Academy.Instance.GetSideChannels<FloatPropertiesChannel>()[0]?

Not sure, if we want to demonstrate the use of `GetSideChannels` we would need to have a way to distinguish the FloatPropertiesChannel obtained and we do not have way to do this...

File: com.unity.ml-agents/Runtime/Academy.cs (hunk 1 above)
Comment (vincentpierre) [reply com.unity.ml-agents/Runtime/Academy.cs]: I am okay with removing it. But we are not 100% sure next release is going to be 1.x are we?

**Comment by chriselion** [comment]
> Not sure, if we want to demonstrate the use of GetSideChannels we would need to have a way to distinguish the FloatPropertiesChannel obtained and we do not have way to do this...

If we don't have one, neither do our users.

File: com.unity.ml-agents/CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 ### Major Changes
+ - Introduced the `SideChannelUtils` to register, unregister and access side channels.
```
Comment (chriselion) [inline com.unity.ml-agents/CHANGELOG.md]: Can you update the "Added a feature to allow sending stats..." below too? It still references Academy.Instance. GetSideChannel

File: docs/Custom-SideChannels.md [hunk 3]
```diff
@@ -123,15 +123,15 @@ public class RegisterStringLogSideChannel : MonoBehaviour
         Application.logMessageReceived += stringChannel.SendDebugStatementToPython;
 
         // The channel must be registered with the Academy
-        Academy.Instance.RegisterSideChannel(stringChannel);
+        SideChannelUtils.RegisterSideChannel(stringChannel);
```
Comment (chriselion) [inline docs/Custom-SideChannels.md]: Update comment to remove Academy

File: docs/Training-Environment-Parameter-Randomization.md [hunk 4]
```diff
@@ -20,7 +20,7 @@ Ball scale of 0.5          |  Ball scale of 4
 
 
 To enable variations in the environments, we implemented `Environment Parameters`.
-`Environment Parameters` are `Academy.Instance.FloatProperties` that can be read when setting
+`Environment Parameters` are values in the Academy's `FloatPropertiesChannel` that can be read when setting
```
Comment (chriselion) [inline docs/Training-Environment-Parameter-Randomization.md]: Not really the Academy's anymore.

**Pull request merged**